[BryanCastorius] iP

BryanCastorius opened the pull request.

BryanCastorius committed:
Add level 0

BryanCastorius committed:
Add level 1

BryanCastorius committed:
Add level 2

BryanCastorius committed:
Add level 3

BryanCastorius committed:
Update Task.java

Indentation fix

BryanCastorius committed:
Update Task.java

Indentation fix

BryanCastorius committed:
Add level 4

BryanCastorius committed:
Updated Task.java indentation

tannerlie commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                    listTasks[indexToUnmark - 1].unmarkAsDone();
+                    System.out.println(LINE);
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println(listTasks[indexToUnmark - 1].getStatusIcon() + " " + listTasks[indexToUnmark - 1].getDescription());
```
line length should not be longer than 120 characters. Consider shortening or wrapped lines

Heart-of-N1 commented on src/main/java/Event.java:
Good job on using inheritance to simply the codes! :)

Heart-of-N1 commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                    }
+
+                } else {
+                    System.out.println(LINE);
+                    System.out.println("List is full. Cannot add more items.");
+                    System.out.println(LINE);
```
Creating a Printing class that can be inherited by other classes can be a potential solution to reudce code duplication.

Heart-of-N1 commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                            System.out.println(LINE);
+                        }
+                        else {
+                            String removeEventString = text.replaceFirst("event", "").trim();
+                            String[] firstPartition = removeEventString.split("/from");
+                            String eventItem = firstPartition[0];
+                            String fromAndToString = firstPartition[1];
+                            String[] secondPartition = fromAndToString.split("/to");
+                            String eventFrom = secondPartition[0];
+                            String eventTo = secondPartition[1];
+
+                            listTasks[insertIndex] = new Event (eventItem, eventFrom, eventTo);
+                            System.out.println(LINE);
+                            System.out.println("Got it. I've added this task:");
+                            System.out.println("  " + listTasks[insertIndex]);
+                            insertIndex += 1;
+                            System.out.println("Now you have " + insertIndex + " tasks in the list.");
+                            System.out.println(LINE);
```
The chunk of code here can benefit from refactoring to increase code readability.

Heart-of-N1 commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+            if (text.equals("bye")) {
+                System.out.println(LINE);
+                return;
+            } else if (text.equals("list")) {
```
Using enumerations for the task types can help reduce the number of if-else statements.

tannerlie commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                            System.out.println("Invalid command.");
+                            System.out.println(LINE);
+                        }
+                        else {
```
wrong layout for else statement. Place it on the same line as the closing bracket

tannerlie commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                        System.out.println("Now you have " + insertIndex + " tasks in the list.");
+                        System.out.println(LINE);
+                    }
+                    else if (text.startsWith("deadline ")) {
```
wrong layout for else statement. Place it on the same line as the closing bracket

tannerlie commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,183 @@
+                            System.out.println("Invalid command.");
+                            System.out.println(LINE);
+                        }
+                        else {
```
wrong layout for else statement. Place it on the same line as the closing bracket

BryanCastorius committed:
Code refactoring

BryanCastorius committed:
Add Invalid Keyword Exception

BryanCastorius committed:
Refactor exception messages

BryanCastorius committed:
Add missing description exception

BryanCastorius committed:
Add invalid task index exception

BryanCastorius committed:
Merge level 5

BryanCastorius committed:
Merge branch 'master' into branch-level-5

BryanCastorius committed:
Revert "Add invalid task index exception"

This reverts commit 2696144ec659ad713a40cf57942c444446d44e9c.

BryanCastorius committed:
Merge branch 'master' of https://github.com/BryanCastorius/ip

BryanCastorius committed:
Revert "Add invalid task index exception"

This reverts commit 2696144ec659ad713a40cf57942c444446d44e9c.

BryanCastorius committed:
Merge branch 'branch-level-5'

BryanCastorius committed:
Revert "Revert "Add invalid task index exception""

This reverts commit 5c70ce88a2067de74c431b06a44edf92ad7cfd05.

BryanCastorius committed:
Merge branch 'branch-level-5'

BryanCastorius committed:
Create packages

BryanCastorius committed:
Change import

BryanCastorius committed:
Create collections

BryanCastorius committed:
Add delete

BryanCastorius committed:
Add save functionality

BryanCastorius committed:
Merge branch 'branch-level-7'

BryanCastorius committed:
Add auto load

BryanCastorius committed:
Create JAR file

wenxin-c commented on src/main/java/gandalf/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+            } else if (hasSaidDelete(userInput)) {
+                deleteUserTasks(userInput, listTasks);
+            }
+            else {
```
Please be consistent with the bracket style.

wenxin-c commented on src/main/java/gandalf/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+
+    private static void deleteUserTasks(String userInput, ArrayList<Task> listTasks)
+            throws InvalidTaskDeletionException {
+        if (listTasks.isEmpty() || userInput.trim().length() == 6) {
```
Try not to use magic literals(e.g. magic number 6 at this line, magic strings below), can consider giving it a meaning by defining a constant and use the constant in the code instead.

wenxin-c commented on src/main/java/gandalf/Gandalf.java:
```diff
@@ -0,0 +1,212 @@
+        startMessage();
+        startProgram();
+        saveTasks();
+        endMessage();
```
Good abstraction!!

wenxin-c commented on src/main/java/gandalf/Gandalf.java:
```diff
@@ -0,0 +1,212 @@
+
+    private static void loadTasks(String filePath)
+            throws FileEmptyException, FileNotFoundException {
+        File saveFile = new File(filePath);
```
Try to name a variable sound like a noun instead of a verb.

wenxin-c commented on src/main/java/gandalf/Gandalf.java:
```diff
@@ -0,0 +1,212 @@
+                String text = load.nextLine();
+                String parsedTask = parseTask(text);
+                previousTasks.add(parsedTask);
+                boolean marked = checkTasksMarkings(text);
```
Try to make boolean variable sound more like a boolean, e.g. isMarked.

wenxin-c commented on src/main/java/gandalf/Gandalf.java:
```diff
@@ -0,0 +1,212 @@
+        } catch (FileNotFoundException e) {
+            System.out.println(LINE);
+            System.out.println("File not found. You may start creating your new list.");
+        } catch (FileEmptyException e) {
```
Consider giving the exception a more meaningful name, instead of simply e.

wenxin-c reviewed (commented):
Well done on your ip so far. Overall the code is well written. Some comments is that try to be consistent with your bracket style and avoid using magic literals .e.g magic numbers and magic strings.

BryanCastorius committed:
Remove echo functionality

BryanCastorius committed:
Add Parser class

BryanCastorius committed:
Add Storage, Ui classes

BryanCastorius committed:
Improve code quality

BryanCastorius committed:
Remove input confirmation for loaded tasks

BryanCastorius committed:
Improve parser class functionality

BryanCastorius committed:
Add A-MoreOOP

BryanCastorius committed:
Add level 8

BryanCastorius committed:
Add level 9

BryanCastorius committed:
Merge pull request #1 from BryanCastorius/branch-level-8

Level 8 pull request

BryanCastorius committed:
Merge branch 'master' into branch-level-9

BryanCastorius committed:
Merge pull request #2 from BryanCastorius/branch-level-9

Level 9 Pull Request

BryanCastorius committed:
Fix left out import statement

BryanCastorius committed:
Amend Parser after DateTime format change

BryanCastorius committed:
Refactor Gandalf main page and Ui

BryanCastorius committed:
JavaDoc documentation

BryanCastorius committed:
Add user guide

BryanCastorius committed:
Update README.md

BryanCastorius committed:
Update README.md icon

BryanCastorius committed:
Add more JavaDoc documentation

I did not know that A-JavaDoc is supposed to be a new branch, hence I only made a few updates in this branch.

BryanCastorius committed:
Minor Gandalf icon edit

BryanCastorius committed:
Merge pull request #3 from BryanCastorius/branch-A-JavaDoc

Branch a java doc
